includes/Modules/Subscribe_With_Google/EditPost.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Subscribe_With_Google\EditPost
+ *
+ * @package   Google\Site_Kit\Modules\Subscribe_With_Google
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Subscribe_With_Google;
+
+/** Supports editing of posts. */
+final class EditPost {
+
+	/**
+	 * Name of product field.
+	 *
+	 * @var string
+	 */
+	private $product_field_name;
+
+	/**
+	 * Name of nonce.
+	 *
+	 * @var string
+	 */
+	private $nonce_name;
+
+	/**
+	 * Action of nonce.
+	 *
+	 * @var string
+	 */
+	private $nonce_action;
+
+	/**
+	 * Settings for SwG.
+	 *
+	 * @var object
+	 */
+	private $settings;
+
+	/**
+	 * Adds action handlers.
+	 *
+	 * @param object $settings Settings for SwG.
+	 */
+	public function __construct( $settings ) {
+		$this->product_field_name = Key::from( 'product' );
+		$this->nonce_name         = Key::from( 'edit_post_nonce' );
+		$this->nonce_action       = Key::from( 'saving_post' );
+		$this->settings           = $settings;
+
+		// Render meta box on Post Edit page.
+		add_action( 'add_meta_boxes', array( $this, 'add_meta_boxes' ) );
+
+		// Handle Posts being saved.
+		add_action( 'save_post', array( $this, 'save_post' ) );
```

Comment:
While I'm not going to review these classes further as part of this PR, a general note about this already: We should never add hooks in a class constructor, as it is a side effect that shouldn't be caused from instantiation. In Site Kit, the established approach is to give such a class a `register` method and add the hooks there (see e.g. the `Subscribe_With_Google` class you added). Then, effectively the `register` method of one class needs to call the `register` methods of the other classes it controls.